### PR TITLE
Add package: bluez

### PR DESCRIPTION
**Pull request by WilliamBehrens**

### CODE REVIEW

File: recipes/bluez/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if not self.settings.os == "Linux":
+            ConanException(f"Unable to build BlueZ on {self.settings.os}")
```
Comment (uilianries) [inline recipes/bluez/all/conanfile.py:42]:
```suggestion
            ConanException(f"Unable to build BlueZ on {self.settings.os}")
 
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
         del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
```

File: recipes/bluez/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+    def build(self):
+        autotools = self._config_autotools()
+        autotools.make()
+
+    def package(self):
+        autotools = self._config_autotools()
+        autotools.install()
+
+    def package_info(self):
```
Comment (uilianries) [inline recipes/bluez/all/conanfile.py:90]:
```suggestion
    def package_info(self):
        self.cpp_info.set_property("pkg_config_name", "bluetooth")
        self.cpp_info.set_property("cmake_file_name", "bluez")
        self.cpp_info.set_property("cmake_target_name", "bluez::bluez")
        # TODO: to remove in conan v2 once cmake_find_package* & pkg_config generators removed
```
Get prepared to CMakeDeps

File: recipes/bluez/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        "fPIC": True,
+        "with_usb": False
+    }
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "."
+
```
[the remark this answers is not in the document]
Comment (WilliamBehrens) [reply recipes/bluez/all/conanfile.py]: Then what should I set destination to for 
```python
def source(self):
    tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```

**Comment by uilianries** [comment]
@WilliamBehrens my guess is bluez needs some patch to find the package, otherwise, it will try to use from system only

**Comment by SSE4** [comment]
> I'm unsure why it's not able to find Glib currently?

it would be nice to look at `config.log` for an actual error, it usually contains a more precise hint. @uilianries can you try to build locally on Linux and check if there are more hints in `config.log`?

**Comment by uilianries** [comment]
I just checked the config.log and bluez wants `glib-2.0.pc`, however we provide `glib.pc`. Now we need to discover who is wrong and fix.

**Comment by uilianries** [comment]
I built glib from sources, and indeed the correct name is glib-2.0.pc. We need to fix the recipe.

```
conan@7b28c6a0f48f:/tmp/glib-2.72.2$ ninja -C _build install
ninja: Entering directory `_build'
[0/1] Installing files.
Installing glib/libglib-2.0.so.0.7200.2 to /tmp/install/lib
Installing glib/gtester to /tmp/install/bin
Installing gobject/glib-enumtypes.h to /tmp/install/include/glib-2.0/gobject
Installing gobject/libgobject-2.0.so.0.7200.2 to /tmp/install/lib
Installing gobject/gobject-query to /tmp/install/bin
Installing gthread/libgthread-2.0.so.0.7200.2 to /tmp/install/lib
Installing gmodule/libgmodule-2.0.so.0.7200.2 to /tmp/install/lib
Installing gio/gioenumtypes.h to /tmp/install/include/glib-2.0/gio
Installing gio/libgio-2.0.so.0.7200.2 to /tmp/install/lib
...

Installing /tmp/glib-2.72.2/_build/meson-private/glib-2.0.pc to /tmp/install/lib/pkgconfig
...
conan@7b28c6a0f48f:/tmp/glib-2.72.2$ find /tmp/install -name "*.pc"
/tmp/install/lib/pkgconfig/gio-unix-2.0.pc
/tmp/install/lib/pkgconfig/gobject-2.0.pc
/tmp/install/lib/pkgconfig/gmodule-no-export-2.0.pc
/tmp/install/lib/pkgconfig/gmodule-export-2.0.pc
/tmp/install/lib/pkgconfig/gmodule-2.0.pc
/tmp/install/lib/pkgconfig/gthread-2.0.pc
/tmp/install/lib/pkgconfig/gio-2.0.pc
/tmp/install/lib/pkgconfig/glib-2.0.pc
```

**Comment by uilianries** [comment]
@WilliamBehrens I guess this message comes from autotools itself, because I was not able to find such message in Conan code base: https://github.com/conan-io/conan/search?q=configuration+directory+is+required

**Comment by WilliamBehrens** [comment]
I had guess as much but I posted it here because the `pkg_config` generator will work just fine in most cases but when using PkgConfigDeps that directory error appears. Which leads me to believe it's some kind of incompatibility somewhere

**Comment by uilianries** [comment]
@WilliamBehrens I need to check again, maybe it's related to glib package, which is not well prepared to PkgConfigDeps. I did that PR, but I checked mostly pkg_config.

**Pull request closed without merging**